pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_slice.py

```diff
@@ -0,0 +1,417 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from six import raise_from, add_metaclass
+
+from spinn_utilities.overrides import overrides
+from spinn_utilities.abstract_base import AbstractBase, abstractmethod
+from pacman.model.graphs.machine import MachineEdge
+from pacman.utilities.algorithm_utilities.\
+    partition_algorithm_utilities import get_remaining_constraints
+from pacman.exceptions import PacmanPartitionException, PacmanValueError
+from pacman.model.graphs import AbstractVirtual
+from pacman.model.graphs.common import Slice
+from .abstract_splitter_common import AbstractSplitterCommon
+
+
+@add_metaclass(AbstractBase)
+class AbstractSplitterSlice(AbstractSplitterCommon):
+    """ contains default logic for splitting by slice
+    """
+
+    __slots__ = ["_called"]
+
+    NOT_SUITABLE_VERTEX_ERROR = (
+        "The vertex {} cannot be supported by the {} as "
+        "the vertex does not support the required API of "
+        "LegacyPartitionerAPI. Please inherit from the class in "
+        "pacman.model.partitioner_interfaces.legacy_partitioner_api and try "
+        "again.")
+
+    NO_MORE_RESOURCE_AVAILABLE_ERROR = (
+        "No more of vertex '{}' would fit on the board:\n"
+        "    Allocated so far: {} atoms\n"
+        "    Request for SDRAM: {}\n"
+        "    Largest SDRAM space: {}")
+
+    FAIL_TO_ALLOCATE_RESOURCES = (
+        "Unable to allocate requested resources available to vertex "
+        "'{}':\n{}")
+
+    MACHINE_LABEL = "{}:{}:{}"
+
+    def __init__(self, splitter_name):
+        super(AbstractSplitterSlice, self).__init__(splitter_name)
+        self._called = False
+
+    @overrides(AbstractSplitterCommon.get_out_going_vertices)
+    def get_out_going_vertices(self, edge, outgoing_edge_partition):
```

Comment:
I guess I'm still slightly confused here.  Can you explain in what circumstances get_out_going_vertices(...) and get_in_coming_vertices(...) will return different vertices?

Reply:
On further reading, and with the help of the SplitterDelayVertexSlice in sPyNNaker, I think I now understand what's going on here, but it may still be helpful to clarify this question with an answer anyway.  It may also be useful to see an example of what you expect this to look like (I know you can't run it directly without creating separate aplx binaries) when e.g. there are multiple "synapse vertices" for each "neuron vertex".

Reply:
to add, over and above the behaviour of the delay splitter. If you think of the ear. you will have specific ihcan / aggregator cores that send specific neurons worth of packets. 

But you also need to consider in the compartment model, you dont want to link from the synapse cores to other verts, you only want the neuron cores. and this is your only access point for getting those.